From 2720d63c0cb11cdd96523e3ac22a6d54811ef661 Mon Sep 17 00:00:00 2001
From: Alexander Grund <alexander.grund@tu-dresden.de>
Date: Fri, 10 Jul 2020 09:57:00 +0200
Subject: [PATCH] Define PSIMD_SOURCE_DIR when including FP16

Avoids a superflous redownload when *NNPACK is not used (e.g. on Power)
---
 cmake/Dependencies.cmake | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index d6ffc50660675..6fb5a91338718 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -673,8 +673,13 @@ endif()
 
 # ---[ Caffe2 depends on FP16 library for half-precision conversions
 if (NOT TARGET fp16)
-  if (NOT DEFINED FP16_SOURCE_DIR)
-    set(FP16_SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/../third_party/FP16" CACHE STRING "FP16 source directory")
+  set(CAFFE2_THIRD_PARTY_ROOT "${PROJECT_SOURCE_DIR}/third_party")
+  # PSIMD is required by FP16
+  if(NOT DEFINED PSIMD_SOURCE_DIR)
+    set(PSIMD_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/psimd" CACHE STRING "PSimd source directory")
+  endif()
+  if(NOT DEFINED FP16_SOURCE_DIR)
+    set(FP16_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/FP16" CACHE STRING "FP16 source directory")
   endif()
 
   set(FP16_BUILD_TESTS OFF CACHE BOOL "")
